modules/nlu/src/backend/predict-pipeline.ts

```diff
@@ -464,25 +464,37 @@ async function extractSlots(input: PredictStep, predictors: Predictors): Promise
 }
 
 function MapStepToOutput(step: PredictStep, startTime: number): PredictOutput {
+  const entitiesMapper = (e?: EntityExtractionResult) => {
+    if (!e) {
+      return e
+    }
+
+    return {
+      name: e.type,
+      type: e.metadata.entityId,
+      data: {
+        unit: e.metadata.unit,
+        value: e.value
+      },
+      meta: {
+        sensitive: e.sensitive,
+        confidence: e.confidence,
+        end: e.end,
+        source: e.metadata.source,
+        start: e.start
```

Comment:
If you want to support either start or startPos, you could write it like this: `start: e.start ?? e.startPos`, this way you will avoid to map twice over the same array.

Also, please add a comment why it can be either one. i'm a bit lost

Reply:
If you really wanna know, there's a type clash between type `UtteranceEntity` and `EntityExtractionResult` which are pretty similar but not exactly the same.

I don't want to make to much refactor here, so that's why it can be both.